tests/components/plex/test_config_flow.py

```diff
@@ -502,3 +530,194 @@ async def test_multiple_servers_with_import(hass):
+    assert result["step_id"] == "manual_setup"
+
+    mock_plex_server = MockPlexServer()
+
+    MANUAL_SERVER = {
+        CONF_HOST: MOCK_SERVERS[0][CONF_HOST],
+        CONF_PORT: MOCK_SERVERS[0][CONF_PORT],
+        CONF_SSL: False,
+        CONF_VERIFY_SSL: True,
+        CONF_TOKEN: MOCK_TOKEN,
+    }
+
+    MANUAL_SERVER_NO_HOST_OR_TOKEN = {
+        CONF_PORT: MOCK_SERVERS[0][CONF_PORT],
+        CONF_SSL: False,
+        CONF_VERIFY_SSL: True,
+    }
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"], user_input=MANUAL_SERVER_NO_HOST_OR_TOKEN
+    )
+
+    assert result["type"] == "form"
+    assert result["step_id"] == "manual_setup"
+    assert result["errors"]["base"] == "host_or_token"
+
+    with patch(
+        "plexapi.server.PlexServer", side_effect=requests.exceptions.SSLError,
+    ):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], user_input=MANUAL_SERVER
+        )
+
+    assert result["type"] == "form"
+    assert result["step_id"] == "manual_setup"
+    assert result["errors"]["base"] == "ssl_error"
+
+    with patch(
+        "plexapi.server.PlexServer", side_effect=WrongCertValidaitionException,
+    ):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], user_input=MANUAL_SERVER
+        )
+
+    assert result["type"] == "form"
+    assert result["step_id"] == "manual_setup"
+    assert result["errors"]["base"] == "ssl_error"
+
+    with patch(
+        "homeassistant.components.plex.PlexServer.connect",
+        side_effect=requests.exceptions.SSLError,
+    ):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], user_input=MANUAL_SERVER
+        )
+
+    assert result["type"] == "form"
+    assert result["step_id"] == "manual_setup"
+    assert result["errors"]["base"] == "ssl_error"
+
+    with patch("plexapi.server.PlexServer", return_value=mock_plex_server), patch(
+        "homeassistant.components.plex.PlexWebsocket.listen"
+    ):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], user_input=MANUAL_SERVER
+        )
+
+    assert result["type"] == "create_entry"
+    assert result["title"] == mock_plex_server.friendlyName
+    assert result["data"][CONF_SERVER] == mock_plex_server.friendlyName
+    assert result["data"][CONF_SERVER_IDENTIFIER] == mock_plex_server.machineIdentifier
+    assert result["data"][PLEX_SERVER_CONFIG][CONF_URL] == mock_plex_server._baseurl
+    assert result["data"][PLEX_SERVER_CONFIG][CONF_TOKEN] == MOCK_TOKEN
+
+
+async def test_manual_config_with_token(hass):
+    """Test creating via manual configuration with only token."""
+
+    result = await hass.config_entries.flow.async_init(
+        config_flow.DOMAIN, context={"source": "user", "show_advanced_options": True}
+    )
+
+    assert result["type"] == "form"
+    assert result["step_id"] == "user_advanced"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5462445)